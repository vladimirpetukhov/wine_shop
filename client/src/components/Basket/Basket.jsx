import React from 'react';
import Badge from '@material-ui/core/Badge';
import IconButton from '../buttons/IconButton/IconButton';
import NavLink from '../NavLink/NavLink';

const Basket = (props) => {
  const cart = JSON.parse(window.localStorage.getItem('cart'));
  const basket = (cart || []).reduce(
    (acc, curr) => {
      const count = acc.count + curr.productQuantityCart;
      const price = acc.price + curr.price * curr.productQuantityCart;
      return { count, price };
    },
    { count: 0, price: 0 }
  );

  return (
    <NavLink to={'/cart'}>
      <Badge badgeContent={basket.count} color="secondary">
        <IconButton icon="basket" attr={{ color: 'inherit' }} />
      </Badge>
      <p> - {basket.price.toFixed(2)} лв.</p>
    </NavLink>
  );
};

export default Basket;
