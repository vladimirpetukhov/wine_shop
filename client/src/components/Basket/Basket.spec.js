import React from 'react';
import Basket from './Basket';
import renderer from 'react-test-renderer';
import { BrowserRouter } from 'react-router-dom';
test('Component "Basket" is mount', () => {
  const component = renderer.create(
    <BrowserRouter>
      <Basket />
    </BrowserRouter>
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
