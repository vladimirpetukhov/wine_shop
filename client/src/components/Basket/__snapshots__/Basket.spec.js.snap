// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component "Basket" is mount 1`] = `
<a
  href="/cart"
  onClick={[Function]}
>
  <span
    className="MuiBadge-root"
  >
    <button
      className="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit"
      disabled={false}
      onBlur={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      <span
        className="MuiIconButton-label"
      >
        <svg
          aria-hidden={true}
          className="MuiSvgIcon-root"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M17.21 9l-4.38-6.56c-.19-.28-.51-.42-.83-.42-.32 0-.64.14-.83.43L6.79 9H2c-.55 0-1 .45-1 1 0 .09.01.18.04.27l2.54 9.27c.23.84 1 1.46 1.92 1.46h13c.92 0 1.69-.62 1.93-1.46l2.54-9.27L23 10c0-.55-.45-1-1-1h-4.79zM9 9l3-4.4L15 9H9zm3 8c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
          />
        </svg>
      </span>
    </button>
    <span
      className="MuiBadge-badge MuiBadge-anchorOriginTopRightRectangle MuiBadge-colorSecondary MuiBadge-invisible"
    >
      0
    </span>
  </span>
  <p>
     - 
    0.00
     лв.
  </p>
</a>
`;
